app/init.php

```diff
@@ -1083,6 +1095,39 @@ function (mixed $value) {
     return $database;
 }, ['pools', 'cache']);
 
+App::setResource('getProjectDB', function (Group $pools, Database $dbForConsole, $cache) {
+    $databases = []; // TODO: @Meldiron This should probably be responsibility of utopia-php/pools
+
+    $getProjectDB = function (Document $project) use ($pools, $dbForConsole, $cache, &$databases) {
+        if ($project->isEmpty() || $project->getId() === 'console') {
+            return $dbForConsole;
+        }
+
+        $databaseName = $project->getAttribute('database');
+
+        if (isset($databases[$databaseName])) {
+            $database = $databases[$databaseName];
+            $database->setNamespace('_' . $project->getInternalId());
+            return $database;
+        }
+
+        $dbAdapter = $pools
+            ->get($databaseName)
+            ->pop()
+            ->getResource();
+
+        $database = new Database($dbAdapter, $cache);
+
+        $databases[$databaseName] = $database;
+
+        $database->setNamespace('_' . $project->getInternalId());
+
+        return $database;
+    };
+
+    return $getProjectDB;
+}, ['pools', 'dbForConsole', 'cache']);
```

Comment:
this resource is redundant. We have the `dbForProject` resource already which does the same thing.

We needed the `getProjectDB` resource only in the workers and tasks

Reply:
This includes cache layer that `dbForProject` doesnt. This ensures that if you call it 100 times, and all 100 projects are on same DB, it only gives 1 DB connection instead of 100 connections. That prevents "pool is empty" in some scenarios.

We implement it together like this some months ago.

Is pools library now ready to remove this? Are we sure we wont get "pool is empty" problems anymore?